src/rdf4cpp/rdf/Literal.cpp

```diff
@@ -1281,10 +1282,59 @@ bool Literal::is_string_like() const noexcept {
         return false;
     }
 
+    // TODO: this does not cover custom types deriving from xsd:string or xsd:langString
     return type == datatypes::xsd::String::fixed_id || type == datatypes::rdf::LangString::fixed_id;
 }
 
 Literal Literal::add(Literal const &other, Node::NodeStorage &node_storage) const noexcept {
+    // DayTimeDuration + DayTimeDuration
+    {
+        auto this_v = this->cast_to_supertype_value<datatypes::xsd::DayTimeDuration>();
+        auto other_v = other.cast_to_supertype_value<datatypes::xsd::DayTimeDuration>();
+        if (this_v.has_value() && other_v.has_value()) {
+            auto r = datatypes::registry::util::to_checked(*this_v) + datatypes::registry::util::to_checked(*other_v);
+            if (r.count().is_invalid())
+                return Literal{};
+
+            return make_typed_from_value<datatypes::xsd::DayTimeDuration>(datatypes::registry::util::from_checked(r));
+        }
+    }
+    // YearMonthDuration + YearMonthDuration
+    {
+        auto this_v = this->cast_to_supertype_value<datatypes::xsd::YearMonthDuration>();
+        auto other_v = other.cast_to_supertype_value<datatypes::xsd::YearMonthDuration>();
+        if (this_v.has_value() && other_v.has_value()) {
+            auto r = datatypes::registry::util::to_checked(*this_v) + datatypes::registry::util::to_checked(*other_v);
+            if (r.count().is_invalid())
+                return Literal{};
+
+            return make_typed_from_value<datatypes::xsd::YearMonthDuration>(datatypes::registry::util::from_checked(r));
+        }
+    }
+    // DateTime(&Subclasses) + Duration(&Subclasses)
```

Comment:
this covers more cases than the ones defined in the standard, right? the standard defines (dayTimeDuration+time), (dayTimeDuration+date), (yearMonthDuration+time), and (yearMonthDuration+time).

Reply:
Yes, my reasoning for that is: the algorythm to do the calculation is defined and it simplifies code to not differentiate between the 2 cases.